### PR TITLE
feat: Adding Autoscaling, Observability, and Beam(python) modules

### PR DESCRIPTION
**Pull request by senkinnar**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

Added support for Flink Operator Autoscaling, Observability and an example to run Beam python jobs using Flink Operator.

### More

- [*] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [ ] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [ ] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Review by ovaleanu — COMMENTED** [review]
Please modify the PR title. Available types are:
 - feat: A new feature
 - fix: A bug fix
 - docs: Documentation only changes
 - style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
 - refactor: A code change that neither fixes a bug nor adds a feature
 - perf: A code change that improves performance
 - test: Adding missing tests or correcting existing tests
 - build: Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
 - ci: Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
 - chore: Other changes that don't modify src or test files
 - revert: Reverts a previous commit

**Review by ovaleanu — COMMENTED** [review]
Please add back the Prometheus and Grafana stack parameters which includes also having AMP as a data source. You can check the other blueprints

**Comment by senkinnar** [comment]
Added Prometheus and Grafana stack parameters and changed name of the PR

**Review by ovaleanu — CHANGES_REQUESTED** [review]
Change default value for variable "enable_amazon_prometheus" to true. Everything else LGTM. :rocket:

**Review by ovaleanu — COMMENTED** [review]
Default variable "enable_amazon_prometheus" should to true. Thank you!

File: streaming/flink/helm-values/kube-prometheus.yaml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 prometheus:
   prometheusSpec:
+    serviceMonitorSelectorNilUsesHelmValues: false
+    podMonitorSelectorNilUsesHelmValues: false
+    serviceMonitorSelector: { }
```
Comment (vara-bonthu) [inline streaming/flink/helm-values/kube-prometheus.yaml:5]: extra space between curly braces. Is that ok? one below as well

File: streaming/flink/helm-values/prom-grafana-values.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  # -- Additional plugins to be installed during Grafana startup,
+  # `grafana-polystat-panel` is used by the default Cassandra dashboards.
+  plugins:
+    - grafana-polystat-panel
+  grafana.ini: {}
+  image:
+    repository: grafana/grafana
+    tag: 7.5.11
+    sha: ""
+    pullPolicy: IfNotPresent
```
Comment (vara-bonthu) [inline streaming/flink/helm-values/prom-grafana-values.yaml:21]: This file is not referenced in this PR. Do we need this?

File: streaming/flink/variables.tf [hunk 3]
```diff
@@ -40,15 +40,15 @@ variable "private_subnets" {
   default     = ["10.1.0.0/17", "10.1.128.0/18"]
   type        = list(string)
 }
-
+#We are using kube_prometheus_stack here
```
Comment (vara-bonthu) [inline streaming/flink/variables.tf:43]: remove this comment line

File: streaming/flink/variables.tf [hunk 4]
```diff
@@ -40,15 +40,15 @@ variable "private_subnets" {
...
 variable "enable_amazon_prometheus" {
   description = "Enable AWS Managed Prometheus service"
   type        = bool
-  default     = true
+  default     = false
 }
-
+# We are not using Yunikorn here
```
Comment (vara-bonthu) [inline streaming/flink/variables.tf:49]: remove this comment line

File: streaming/flink/helm-values/prom-grafana-values.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+prometheus:
+  prometheusSpec:
+    serviceMonitorSelectorNilUsesHelmValues: false
+    podMonitorSelectorNilUsesHelmValues: false
+    serviceMonitorSelector: {}
+    serviceMonitorNamespaceSelector: {}
+grafana:
```
Comment (vara-bonthu) [inline streaming/flink/helm-values/prom-grafana-values.yaml:7]: Did you mean to update this config in the above file ? `streaming/flink/helm-values/kube-prometheus.yaml` This file will be used as a default while deploying Kube Prometheus stack.

File: streaming/flink/helm-values/kube-prometheus.yaml (hunk 1 above)
Comment (senkinnar) [reply streaming/flink/helm-values/kube-prometheus.yaml:5]: Removed

File: streaming/flink/helm-values/prom-grafana-values.yaml (hunk 2 above)
Comment (senkinnar) [reply streaming/flink/helm-values/prom-grafana-values.yaml:21]: No removed

File: streaming/flink/variables.tf (hunk 3 above)
Comment (senkinnar) [reply streaming/flink/variables.tf:43]: Done

File: streaming/flink/variables.tf (hunk 4 above)
Comment (senkinnar) [reply streaming/flink/variables.tf:49]: Done

File: streaming/flink/helm-values/prom-grafana-values.yaml (hunk 5 above)
Comment (senkinnar) [reply streaming/flink/helm-values/prom-grafana-values.yaml:7]: Not required already there

**Comment by senkinnar** [comment]
> Default variable "enable_amazon_prometheus" should to true. Thank you!

When this variable is enabled to true the podMonitors for Flink is not working.

**Review by vara-bonthu — APPROVED** [review]
LGTM 🚀 Thank you @senkinnar 🎆

**Pull request merged**